Failing to build with

```  
 Compiling specs v0.16.1 (https://github.com/amethyst/specs.git?rev=d4435bdf496cf322c74886ca09dd8795984919b4#d4435bdf)
   Compiling auth-common v0.1.0 (https://gitlab.com/veloren/auth.git?rev=bffb5181a35c19ddfd33ee0b4aedba741aafb68d#bffb5181)
error[E0075]: SIMD vector length must be a power of two
    --> /run/build/veloren/cargo/vendor/vek/src/vec.rs:2857:13
     |
2857 |               pub struct Vec3<T> { pub x:T, pub y:T, pub z:T }
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
3249 | /     vec_impl_all_vecs!{
3250 | |         simd
3251 | |         #[repr(simd)]
3252 | |     }
     | |_____- in this macro invocation
     |
     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

   Compiling gilrs-core v0.2.6
error[E0075]: SIMD vector length must be a power of two
    --> /run/build/veloren/cargo/vendor/vek/src/vec.rs:3076:13
     |
3076 |               pub struct Extent3<T> { pub w:T, pub h:T, pub d:T }
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
3249 | /     vec_impl_all_vecs!{
3250 | |         simd
3251 | |         #[repr(simd)]
3252 | |     }
     | |_____- in this macro invocation
     |
     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0075]: SIMD vector length must be a power of two
    --> /run/build/veloren/cargo/vendor/vek/src/vec.rs:3166:13
     |
3166 |               pub struct Rgb<T> { pub r:T, pub g:T, pub b:T }
     |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
3249 | /     vec_impl_all_vecs!{
3250 | |         simd
3251 | |         #[repr(simd)]
3252 | |     }
     | |_____- in this macro invocation
     |
     = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

   Compiling smithay-clipboard v0.6.1
   Compiling smithay-clipboard v0.5.2
   Compiling assets_manager v0.4.2
   Compiling cranelift-codegen-meta v0.68.0
error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0075`.
error: could not compile `vek`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: module veloren: Child process exited with code 101
```

My current guess is veloren's build depends on vek 0.11.2 and 0.12.0, but the flatpak only downloads 0.11.2. I can reproduce the flatpak issue locally - but my "pure" git build of veloren works fine.

--- a/net.veloren.veloren.yaml
+++ b/net.veloren.veloren.yaml
@@ -2,8 +2,6 @@ app-id: net.veloren.veloren
 runtime: org.freedesktop.Platform
 runtime-version: '20.08'
 sdk: org.freedesktop.Sdk
-sdk-extensions:
-  - org.freedesktop.Sdk.Extension.rust-nightly
 command: veloren-voxygen
 finish-args:
   - --device=all
@@ -13,9 +11,36 @@ finish-args:
   - --socket=pulseaudio
   - --socket=x11
   - --socket=wayland
-build-options:
-  append-path: /usr/lib/sdk/rust-nightly/bin
 modules:
+  - name: rust-nightly
+    build-options:
+      no-debuginfo: true
+    buildsystem: simple
+    build-commands:
+      - ./install.sh --prefix=/app --without=rust-docs --disable-ldconfig --verbose
+    cleanup:
+      - '*'
+    sources:
+      - type: archive
+        only-arches: [x86_64]
+        url: https://static.rust-lang.org/dist/2021-01-01/rust-nightly-x86_64-unknown-linux-gnu.tar.xz
+        sha256: 7b7139372619ce67914d99d96dfedbc8d33105dbb05f7c5c60d3c5a9ea2be4e0
+        x-checker-data:
+          type: html
+          url: https://gitlab.com/veloren/veloren/-/raw/master/rust-toolchain
+          version-pattern: nightly-(\d\d\d\d-\d\d-\d\d)
+          url-template: https://static.rust-lang.org/dist/$version/rust-nightly-x86_64-unknown-linux-gnu.tar.xz
+
+      - type: archive
+        only-arches: [aarch64]
+        url: https://static.rust-lang.org/dist/2021-01-01/rust-nightly-aarch64-unknown-linux-gnu.tar.xz
+        sha256: e5ebff07432600319ced4909e1fdd5aa45802f4f21a63f0b350f6b98fd0c2054
+        x-checker-data:
+          type: html
+          url: https://gitlab.com/veloren/veloren/-/raw/master/rust-toolchain
+          version-pattern: nightly-(\d\d\d\d-\d\d-\d\d)
+          url-template: https://static.rust-lang.org/dist/$version/rust-nightly-aarch64-unknown-linux-gnu.tar.xz
+
   - name: veloren
     build-options:
       env:

--- a/net.veloren.veloren.yaml
+++ b/net.veloren.veloren.yaml
@@ -15,12 +15,13 @@ finish-args:
   - --socket=wayland
 build-options:
   append-path: /usr/lib/sdk/rust-nightly/bin
-  env:
-    RUSTFLAGS: --remap-path-prefix =../
-    CARGO_HOME: /run/build/veloren/cargo
-    VELOREN_USERDATA_STRATEGY: system
 modules:
   - name: veloren
+    build-options:
+      env:
+        RUSTFLAGS: --remap-path-prefix =../
+        CARGO_HOME: /run/build/veloren/cargo
+        VELOREN_USERDATA_STRATEGY: system
     buildsystem: simple
     build-commands:
       # Unoptimize dev/debug builds to speed up build process

